Updated install, dev and contributing docs (#4378)
sunpy/sunpy: docs/dev_guide/code_standards.rst

Formatting
==========

> We enforce a minimum level of code style with our continuous intergration (the name is ``sunpy.sunpy (python_codestyle [linux]``).

Reviewer: The new stuff in the brackets makes no sense to me.

Author: You mean the name?

Reviewer: Yeah, what does "the name is..." mean?

Author: I will fix that.

Author: See if you think it is clearer now.

> We enforce a minimum level of code style with our continuous intergration (the name is ``sunpy.sunpy (python_codestyle [linux]``).

Author: ```suggestion
We enforce a minimum level of code style with our continuous integration (the check on a pull request will be called ``sunpy.sunpy (python_codestyle [linux]``).
```

This runs a tool called `pre-commit <https://pre-commit.com/>`__.

The settings and tools we use for the pre-commit can be found in the file `.pre-commit-config.yaml` at the root of the sunpy git repository.
Some of the checks are:

> * Check (but will not fix) for various PEP8 issues with flake8.

Reviewer: ```suggestion
* Checks (but doesn't fix) various PEP8 issues with flake8.
```

    +What you will see
    +
    If you want to setup the pre-commit locally, you can do the following::

        $ pip install pre-commit
    diff --git a/docs/dev_guide/documentation.rst b/docs/dev_guide/documentation.rst
    index 5cd914047..b1017f77a 100644
    --- a/docs/dev_guide/documentation.rst
    +++ b/docs/dev_guide/documentation.rst
    @@ -39,9 +39,9 @@ If there are multiple code elements with the same name (e.g. ``peek()`` is a met

    .. code-block:: rst

    -    `GenericMap.peek` or `CompositeMap.peek`
    +    `.GenericMap.peek` or `.CompositeMap.peek`

    -These will show up as `GenericMap.peek` or `CompositeMap.peek`.
    +These will show up as `.GenericMap.peek` or `.CompositeMap.peek`.
    To still show only the last segment you can add a tilde as prefix:

    ERROR: InvocationError for command /home/nabil/GitHub/sunpy/.tox/codestyle/bin/pre-commit run --verbose --all-files --show-diff-on-failure (exited with code 1)
    ___________________________________________________________________________________________ summary ___________________________________________________________________________________________
    ERROR:   codestyle: commands failed

> This will inform you of what checks failed and why, what changes (if any) the command has made to your code.

Reviewer: ```suggestion
This will inform you of what checks failed and why, and what changes (if any) the command has made to your code.
```

If you want to setup the pre-commit locally, you can do the following::

    $ pip install pre-commit

Now you can do::

>     $ pre-commit run --all-files

Author: This is fix you don't want to run tox. Do we want to drop that?

>     $ pre-commit run --all-files

Reviewer 2: ah, maybe just make that a little clearer in the text?

>     $ pre-commit run --all-files

Author: Have a check now?

which will run the tools on all files in the sunpy git repository.

> The pre-commit tools can change some of the files, in other cases it will report problems but will require manual correction.

Reviewer: ```suggestion
The pre-commit tools can change some of the files, but in other cases it will report problems that require manual correction.
```

Formatting
==========

We enforce a minimum level of code style with our continuous intergration (the name is ``sunpy.sunpy (python_codestyle [linux]``).
This runs a tool called `pre-commit <https://pre-commit.com/>`__.

Instead of installing this, you can use "tox" (which is used to run the sunpy test suite) to run these tools without having to setup anything within your own Python virtual environment::

    $ tox -e codestyle

> 

Reviewer 2: After this command I would put an explanation of what will happen when you run this, i.e. it will make changes in the source to auto fix some issue etc (i.e. the bit lower down).

I would then move the install pre-commit instructions down under the "automate this" section?

Author: Added.
